sql/src/main/java/io/crate/action/sql/BatchConsumerToResultReceiver.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.action.sql;
+
+import io.crate.data.BatchConsumer;
+import io.crate.data.BatchIterator;
+import io.crate.data.Row;
+import io.crate.data.RowBridging;
+import io.crate.exceptions.JobKilledException;
+
+import javax.annotation.Nullable;
+
+public class BatchConsumerToResultReceiver implements BatchConsumer {
+
+    private ResultReceiver resultReceiver;
+    private int maxRows;
+    private long rowCount = 0;
+    private BatchIterator activeIt;
+
+    public BatchConsumerToResultReceiver(ResultReceiver resultReceiver, int maxRows) {
+        this.resultReceiver = resultReceiver;
+        this.maxRows = maxRows;
+    }
+
+    @Override
+    public void accept(BatchIterator iterator, @Nullable Throwable failure) {
+        if (failure == null) {
+            consumeIt(iterator);
+        } else {
+            if (iterator != null) {
```

Comment:
As discussed I'll leave it for now and later try if it would simplify things if it was nonNull.

If I mark it as Nullable we'd have to add checks like follows to prevent invalid nullable warnings

```
if (failure == null) {
   ...
} else if (batchIterator != null) {
  ...
} else {
  ???
```